crates/iota-types/src/mock_checkpoint_builder.rs

```diff
@@ -143,13 +161,17 @@ impl MockCheckpointBuilder {
             epoch,
             sequence_number: self
                 .previous_checkpoint
-                .sequence_number
-                .checked_add(1)
-                .unwrap(),
-            network_total_transactions: self.previous_checkpoint.network_total_transactions
+                .as_ref()
+                .map(|c| c.sequence_number + 1)
+                .unwrap_or_default(),
```

Comment:
What is the default value for a `VerifiedCheckpoint`?

Reply:
If the previous checkpoint is absent, it should default to `0`. To make this more I changed it to use `unwrap_or(0)`: https://github.com/iotaledger/iota/pull/7868/commits/acd6e93e738ae4fde633335055e049c87bfd4f68